pkg/common-controller/snapshot_controller.go

```diff
@@ -1149,28 +1148,20 @@ func (ctrl *csiSnapshotCommonController) isVolumeBoundToClaim(volume *v1.Persist
 	return true
 }
 
-func (ctrl *csiSnapshotCommonController) getStorageClassFromVolumeSnapshot(snapshot *crdv1.VolumeSnapshot) (*storagev1.StorageClass, error) {
-	// Get storage class from PVC or PV
-	pvc, err := ctrl.getClaimFromVolumeSnapshot(snapshot)
+// pvDriverFromSnapshot is a helper function to get the CSI driver name from the targeted PersistentVolume.
+// It looks up the PVC from which the snapshot is specified to be created from, and looks for the PVC's corresponding
+// PV. Bi-directional binding will be verified between PVC and PV before the PV's CSI driver is returned.
+// For an non-CSI volume, it returns an error immediately as it's not supported.
+func (ctrl *csiSnapshotCommonController) pvDriverFromSnapshot(snapshot *crdv1.VolumeSnapshot) (string, error) {
+	pv, err := ctrl.getVolumeFromVolumeSnapshot(snapshot)
 	if err != nil {
-		return nil, err
-	}
-	storageclassName := *pvc.Spec.StorageClassName
-	if len(storageclassName) == 0 {
-		volume, err := ctrl.getVolumeFromVolumeSnapshot(snapshot)
-		if err != nil {
-			return nil, err
-		}
-		storageclassName = volume.Spec.StorageClassName
-	}
-	if len(storageclassName) == 0 {
-		return nil, fmt.Errorf("cannot figure out the snapshot class automatically, please specify one in snapshot spec")
+		return "", err
 	}
-	storageclass, err := ctrl.client.StorageV1().StorageClasses().Get(context.TODO(), storageclassName, metav1.GetOptions{})
-	if err != nil {
-		return nil, err
+	// supports ONLY CSI volumes
+	if pv.Spec.PersistentVolumeSource.CSI == nil {
+		return "", fmt.Errorf("snapshot an non-CSI volume is not supported, snapshot:%s/%s", snapshot.Namespace, snapshot.Name)
```

Comment:
nit: snapshotting non-CSI volumes are not supported*

Reply:
done